mint-auditor/src/db/models/gnosis_safe_deposit.rs

```diff
@@ -119,4 +122,188 @@ impl GnosisSafeDeposit {
+                    .filter(
+                        audited_mints::gnosis_safe_deposit_id
+                            .nullable()
+                            .eq(gnosis_safe_deposits::id),
+                    ),
+            )))
+            .load(conn)?)
+    }
+
+    /// Attempt to find a [GnosisSafeDeposit] that has a given nonce and no
+    /// matching entry in the `audited_mints` table.
+    pub fn find_unaudited_deposit_by_nonce(
+        nonce_hex: &str,
+        conn: &Conn,
+    ) -> Result<Option<Self>, Error> {
+        Ok(gnosis_safe_deposits::table
+            .filter(gnosis_safe_deposits::expected_mc_mint_tx_nonce_hex.eq(nonce_hex))
+            .filter(not(exists(
+                audited_mints::table
+                    .select(audited_mints::gnosis_safe_deposit_id)
+                    .filter(
+                        audited_mints::gnosis_safe_deposit_id
+                            .nullable()
+                            .eq(gnosis_safe_deposits::id),
+                    ),
+            )))
+            .first(conn)
+            .optional()?)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::db::{
+        models::{AuditedMint, MintTx},
+        test_utils::{create_gnosis_safe_deposit, insert_gnosis_deposit, TestDbContext},
+    };
+    use mc_common::logger::{test_with_logger, Logger};
+    use mc_transaction_core::TokenId;
+    use mc_transaction_core_test_utils::{create_mint_config_tx_and_signers, create_mint_tx};
+
+    #[test_with_logger]
+    fn test_find_unaudited_deposits_by_nonce(logger: Logger) {
+        let mut rng = mc_util_test_helper::get_seeded_rng();
+        let test_db_context = TestDbContext::default();
+        let mint_auditor_db = test_db_context.get_db_instance(logger.clone());
+        let token_id1 = TokenId::from(1);
+        let conn = mint_auditor_db.get_conn().unwrap();
+
+        // Create gnosis deposits.
+        let mut deposit1 = create_gnosis_safe_deposit(100, &mut rng);
+        let mut deposit2 = create_gnosis_safe_deposit(200, &mut rng);
+
+        // Create two MintTxs.
+        let (_mint_config_tx1, signers1) = create_mint_config_tx_and_signers(token_id1, &mut rng);
+        let mut mint_tx1 = create_mint_tx(token_id1, &signers1, 100, &mut rng);
+        let mut mint_tx2 = create_mint_tx(token_id1, &signers1, 100, &mut rng);
+
+        mint_tx1.prefix.nonce = hex::decode(&deposit1.expected_mc_mint_tx_nonce_hex()).unwrap();
+        mint_tx2.prefix.nonce = hex::decode(&deposit2.expected_mc_mint_tx_nonce_hex()).unwrap();
+
+        let sql_mint_tx1 = MintTx::insert_from_core_mint_tx(0, None, &mint_tx1, &conn).unwrap();
+        let sql_mint_tx2 = MintTx::insert_from_core_mint_tx(0, None, &mint_tx2, &conn).unwrap();
+
+        // Since they haven't been inserted yet, they should not be found.
+        assert!(GnosisSafeDeposit::find_unaudited_deposit_by_nonce(
+            &hex::encode(&mint_tx1.prefix.nonce),
```

Comment:
nit: might consider storing off the hex versions of the nonces and re-using here instead of re-generating throughout test.
```rust
        let nonce1 = deposit1.expected_mc_mint_tx_nonce_hex();
        let nonce2 = deposit2.expected_mc_mint_tx_nonce_hex();
        mint_tx1.prefix.nonce = hex::decode(nonce1).unwrap();
        mint_tx2.prefix.nonce = hex::decode(nonce2).unwrap();
        
        let sql_mint_tx1 = MintTx::insert_from_core_mint_tx(0, None, &mint_tx1, &conn).unwrap();
        let sql_mint_tx2 = MintTx::insert_from_core_mint_tx(0, None, &mint_tx2, &conn).unwrap();

        // Since they haven't been inserted yet, they should not be found.
        assert!(GnosisSafeDeposit::find_unaudited_deposit_by_nonce(
            nonce1,
            &conn
        )
        .unwrap()
        .is_none());

```